src/System.Private.CoreLib/System.Private.CoreLib.csproj

```diff
@@ -462,5 +442,10 @@
       <LogicalName>$(MSBuildProjectName).xml</LogicalName>
     </EmbeddedResource>
   </ItemGroup>
+
+  <ItemGroup>
+    <Service Include="{508349b6-6b84-4df5-91f0-309beebad82d}" />
```

Comment:
VS adds this anytime you run or modify the T4 templates, we should just have it in.

Reply:
Is this actually what is breaking the T4 code gen in general - even outside of VS?

Reply:
VS was broken because of the double-includes, not because this was missing. VS just adds this back any time you make a change to a T4 file, so it is better to have it so you aren't fighting against the IDE by constantly removing it from your local changes.

Reply:
Please comment what this service is meant to do.

Reply:
@davkean this sounds like the same bug where test explorer would add these entries. Do you know about this particular one?

Reply:
Added a comment.